pkg/build/builder/cmd/scmauth/util.go

```diff
@@ -7,19 +7,43 @@ import (
 	"path/filepath"
 	"strings"
 
-	builder "github.com/openshift/builder/pkg/build/builder"
+	buildv1 "github.com/openshift/api/build/v1"
+	"github.com/openshift/builder/pkg/build/builder"
 	utillog "github.com/openshift/builder/pkg/build/builder/util/log"
 )
 
 var log = utillog.ToFile(os.Stderr, 2)
 
-func createGitConfig(includePath string, context SCMAuthContext) error {
+func createGitConfig(includePath string, context SCMAuthContext, gitSource *buildv1.GitBuildSource) error {
 	tempDir, err := ioutil.TempDir("", "git")
 	if err != nil {
 		return err
 	}
 	gitconfig := filepath.Join(tempDir, ".gitconfig")
-	content := fmt.Sprintf("[include]\npath = %s\n", includePath)
+	content := ""
+	if len(includePath) > 0 {
+		content = fmt.Sprintf("[include]\npath = %s\n", includePath)
+	}
+	if gitSource != nil {
+		if gitSource.HTTPProxy != nil && len(*gitSource.HTTPProxy) > 0 {
+			content = content + fmt.Sprintf("[http]\nproxy = %s\n", *gitSource.HTTPProxy)
+		}
+		if gitSource.HTTPSProxy != nil && len(*gitSource.HTTPSProxy) > 0 {
+			content = content + fmt.Sprintf("[https]\nproxy = %s\n", *gitSource.HTTPSProxy)
+		}
+		if gitSource.NoProxy != nil && len(*gitSource.NoProxy) > 0 {
+			noProxies := strings.Split(*gitSource.NoProxy, ",")
+			for _, noProxy := range noProxies {
+				if strings.Contains(gitSource.URI, noProxy) {
```

Comment:
Can we convert the URI to a net/url and test just the hostname portion? https://golang.org/pkg/net/url/#URL

Reply:
I'll give it a try in a separate commit and assuming it works, we can iterate over it

Reply:
ok separate commit pushed ... will report back after I've analyzed the pr's e2e's and cluster-bot with proxy e2e